core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1626,6 +1645,80 @@ private ExprValue exprToDays(ExprValue date) {
     return new ExprLongValue(date.dateValue().toEpochDay() + DAYS_0000_TO_1970);
   }
 
+  /**
+   * To_seconds implementation for ExprValue.
+   *
+   * @param date ExprValue of Date/Datetime/Timestamp/String type.
+   * @return ExprValue.
+   */
+  private ExprValue exprToSeconds(ExprValue date) {
+    return new ExprLongValue(
+        date.datetimeValue().toEpochSecond(ZoneOffset.UTC) + DAYS_0000_TO_1970 * SECONDS_PER_DAY);
+  }
+
+  /**
+   * Helper function to determine the correct formatter for date arguments passed in as integers.
+   *
+   * @param dateAsInt is an integer formatted as one of YYYYMMDD, YYMMDD, YMMDD, MMDD, MDD
+   * @return is a DateTimeFormatter that can parse the input.
+   */
+  private DateTimeFormatter getFormatter(int dateAsInt) {
+    if (dateAsInt < 0 || dateAsInt > 99999999) {
+      throw new DateTimeException("Integer argument was out of range");
+    }
+
+    //Check below from YYYYMMDD - MMDD which format should be used
+
+    //Check if dateAsInt is at least 8 digits long
+    if (dateAsInt - 9999999 > 0) {
+      return DATE_FORMATTER_LONG_YEAR;
+    }
+
+    //Check if dateAsInt is at least 6 digits long
+    if (dateAsInt - 99999 > 0) {
+      return DATE_FORMATTER_SHORT_YEAR;
+    }
+
+    //Check if dateAsInt is at least 5 digits long
+    if (dateAsInt - 9999 > 0) {
+      return DATE_FORMATTER_SINGLE_DIGIT_YEAR;
+    }
+
+    //Check if dateAsInt is at least 4 digits long
+    if (dateAsInt - 999 > 0) {
```

Comment:
`999` magic number

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/232/commits/2b37b0d4090aca8b66dbb02ba9a47e0eda5e27eb